Firestore/core/src/local/leveldb_remote_document_cache.cc

```diff
@@ -171,13 +173,18 @@ MutableDocumentMap LevelDbRemoteDocumentCache::GetAll(
 }
 
 MutableDocumentMap LevelDbRemoteDocumentCache::GetAllExisting(
-    DocumentVersionMap&& remote_map) const {
+    DocumentVersionMap&& remote_map,
+    const core::Query& query,
+    const model::OverlayByDocumentKeyMap& mutated_docs) const {
   BackgroundQueue tasks(executor_.get());
   AsyncResults<std::pair<DocumentKey, MutableDocument>> results;
   for (const auto& key_version : remote_map) {
-    tasks.Execute([this, &results, key_version] {
+    tasks.Execute([this, &results, key_version, &query, &mutated_docs] {
```

Comment:
Sorry I didn't find out earlier, could you please also make `key_version` pass by reference?

Reply:
I do not think query can be captured by reference here, somehow it is deleted before its usage here. This is what caused sanitizer failure.

Reply:
I don't understand why `query` would be deleted :(